### PR TITLE
user provider cleanup

### PR DESCRIPTION
**Pull request by lamont-granquist**
needed to be built on top of #5673 so it might be easier to review and merge that one first.

### CODE REVIEW

**Comment by lamont-granquist** [comment]
@chef/client-core more to review

File: lib/chef/provider/user.rb [hunk 1]
```diff
@@ -175,7 +172,7 @@ def action_lock
       end
 
       def action_unlock
-        if check_lock() == true
+        if check_lock == true
```
Comment (thommay) [inline lib/chef/provider/user.rb]: `if check_lock`?

File: lib/chef/provider/user.rb [hunk 2]
```diff
@@ -137,34 +137,31 @@ def action_create
...
+        return unless compare_user
+        converge_by("modify user #{@new_resource.username}") do
+          manage_user
+          Chef::Log.info("#{@new_resource} modified")
         end
       end
 
       def action_lock
-        if check_lock() == false
+        if check_lock == false
```
Comment (thommay) [inline lib/chef/provider/user.rb]: `if !check_lock` ?

**Comment by lamont-granquist** [comment]
yeah fixing those nits are out of scope i think

**Pull request merged**